File: tile/Tile.java
package pencoding.sidescroller.tile;

import java.awt.Graphics;
import java.awt.Rectangle;

import pencoding.sidescroller.main.Controller;
import pencoding.sidescroller.main.Id;

public abstract class Tile {

	public int x, y;
	public int width, height;
	public int velX, velY;
	

	public boolean solid;
	
	public Id id;
	
	public Controller controller;
	
	public Tile(int x, int y, int width, int height, boolean solid, Id id, Controller controller){
		this.x = x;
		this.y = y;
		this.width = width;
		this.height = height;
		this.solid = solid;
		this.id = id;
		this.controller = controller;
	}
	
	public abstract void render(Graphics g);
	public abstract void tick(); 

	public void die(){
		controller.removeTile(this);
	}
	
	public Rectangle getBounds(){
		return new Rectangle(getX(),getY(),width,height);
	}
	
	// --------------------------- Getters and Setters
	
	public Id getId(){
		return id;
	}
	
	public int getX() {
		return x;
	}

	public void setX(int x) {
		this.x = x;
	}

	public int getY() {
		return y;
	}

	public void setY(int y) {
		this.y = y;
	}

	public int getVelX() {
		return velX;
	}

	public void setVelX(int velX) {
		this.velX = velX;
	}

	public int getVelY() {
		return velY;
	}

	public void setVelY(int velY) {
		this.velY = velY;
	}
	
	public boolean isSolid() {
		return solid;
	}

	public int getWidth() {
		return width;
	}

	public void setWidth(int width) {
		this.width = width;
	}

	public int getHeight() {
		return height;
	}

	public void setHeight(int height) {
		this.height = height;
	}

	public void setSolid(boolean solid) {
		this.solid = solid;
	}
	
	//--------------------------------------------------
	
}

File: tile/Tile.js
/* globals createjs */
var Tile = function (x, y, width, height, solid, id) {
    "use strict";

    var velX = 0,
        velY = 0;

    //initialize
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
    this.solid = solid;
    this.id = id;

    this.getBounds = function () {

        var tileGfx = new createjs.Graphics();
        //first 2 params are x,y location
        tileGfx.drawRect(this.getX(), this.getY(), width, height);
        return tileGfx;
    };


    // --------------------------- Getters and Setters

    this.getId = function () {
        return id;
    };

    this.getX = function () {
        return x;
    };

    this.setX = function (x) {
        this.x = x;
    };

    this.getY = function () {
        return y;
    };

    this.setY = function (y) {
        this.y = y;
    };

    this.getVelX = function () {
        return velX;
    };

    this.setVelX = function (velX) {
        this.velX = velX;
    };

    this.getVelY = function () {
        return velY;
    };

    this.setVelY = function (velY) {
        this.velY = velY;
    };

    this.isSolid = function () {
        return solid;
    };

    this.getWidth = function () {
        return width;
    };

    this.setWidth = function (width) {
        this.width = width;
    };

    this.getHeight = function () {
        return height;
    };

    this.setHeight = function (height) {
        this.height = height;
    };

    this.setSolid = function (solid) {
        this.solid = solid;
    };

};